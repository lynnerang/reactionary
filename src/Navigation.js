import React from 'react';
import { NavLink } from 'react-router-dom';
import './Navigation.scss';

function Navigation(props) {
    return (
      <nav className='App-navigation'>
          <NavLink className='nav-link' exact={true} to='/'><i className="fas fa-home"></i>Home</NavLink>
          <NavLink className='nav-link' to='/play'><i className="fas fa-play"></i>Play</NavLink>
          <NavLink className='nav-link' to='/mycards'><i className="far fa-copy"></i>My Cards</NavLink>
          <NavLink className='nav-link' to='/stats'><i className="fas fa-chart-pie"></i>Stats</NavLink>
          <NavLink className='nav-link' to='/links'><i className="fas fa-link"></i>Links</NavLink>
      </nav>
    )
}

export default Navigation;