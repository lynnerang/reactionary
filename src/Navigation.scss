@import './variables.scss';

.App-navigation {
  background: $dark-gray;
}

.App-navigation {
  height: 100%;
  width: 200px;
  padding-top: 20px;
}

nav {
  display: flex;
  flex-direction: column;
  height: 100%;
}

.nav-link { 
  text-decoration: none;
  color: white;
  padding-bottom: 20px;
  padding-top: 20px;
  padding-left: 25px;
  border-left: 3px solid $dark-gray;
  cursor: pointer;

  .fas, .far, .fas {
    width: 20px;
    font-size: 14px;
    padding-right: 10px;
  }

  &:hover {
    color: $blue;
    border-left: 3px solid $blue;
    background: rgba(255, 255, 255, 0.262);
  }
}

.active {
  color: $secondary-color!important;
  border-left: 3px solid $secondary-color;
  background: rgba(225, 225, 225, 0.16);
} 

