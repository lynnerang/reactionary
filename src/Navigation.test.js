import React from 'react';
import Navigation from './Navigation.js';
import { shallow } from 'enzyme';


describe('Navigation', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow (
      <Navigation 
      />
      )
  });

  it('Should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

});