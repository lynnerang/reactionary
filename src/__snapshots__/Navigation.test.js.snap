// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Should match the snapshot 1`] = `
<nav
  className="App-navigation"
>
  <NavLink
    className="nav-link"
    exact={true}
    to="/"
  >
    <i
      className="fas fa-home"
    />
    Home
  </NavLink>
  <NavLink
    className="nav-link"
    to="/play"
  >
    <i
      className="fas fa-play"
    />
    Play
  </NavLink>
  <NavLink
    className="nav-link"
    to="/mycards"
  >
    <i
      className="far fa-copy"
    />
    My Cards
  </NavLink>
  <NavLink
    className="nav-link"
    to="/stats"
  >
    <i
      className="fas fa-chart-pie"
    />
    Stats
  </NavLink>
  <NavLink
    className="nav-link"
    to="/links"
  >
    <i
      className="fas fa-link"
    />
    Links
  </NavLink>
</nav>
`;
